catalog/plugin/catalog-plugin-metacardingest-network/src/main/java/org/codice/ddf/catalog/plugin/metacard/util/AttributeHelper.java

```diff
@@ -0,0 +1,199 @@
+
+        BundleContext bundleContext = FrameworkUtil.getBundle(AttributeHelper.class)
+                .getBundleContext();
+        serviceReferences.stream()
+                .map((Function<ServiceReference, Object>) bundleContext::getService)
+                .filter(MetacardType.class::isInstance)
+                .map(MetacardType.class::cast)
+                .filter(Objects::nonNull)
+                .map(MetacardType::getAttributeDescriptors)
+                .forEach(descriptors -> loadUniqueDescriptors(descriptors, systemDescriptors));
+    }
+
+    /**
+     * Iterates over each {@link Metacard} in the {@param metacards} collection and applies each
+     * attribute in the {@param attributeMap} only if the attribute is not already set on the
+     * {@link Metacard}. That is, the attribute must be {@code null} or not in the {@link Metacard}'s
+     * map.
+     *
+     * @param metacards    The list of metacards to attempt to add the attributes to.
+     * @param attributeMap The map of attributes to attempt to add. Attributes already set on any
+     *                     given {@link Metacard} will not be changed.
+     */
+    public void applyNewAttributes(List<Metacard> metacards, Map<String, String> attributeMap) {
+        if (metacards.isEmpty() || attributeMap.isEmpty()) {
+            return;
+        }
+
+        Map<String, AttributeDescriptor> systemAndMetacardDescriptors = new HashMap<>();
+        systemAndMetacardDescriptors.putAll(systemDescriptors);
+        metacards.stream()
+                .map(Metacard::getMetacardType)
+                .map(MetacardType::getAttributeDescriptors)
+                .forEach(descriptors -> loadUniqueDescriptors(descriptors,
+                        systemAndMetacardDescriptors));
+
+        metacards.forEach(metacard -> attributeMap.keySet()
+                .stream()
+                .filter(key -> metacard.getAttribute(key) == null)
+                .map(systemAndMetacardDescriptors::get)
+                .filter(Objects::nonNull)
+                .map(ad -> createAttribute(ad, attributeMap.get(ad.getName())))
+                .filter(Objects::nonNull)
+                .forEach(metacard::setAttribute));
+    }
+
+    /**
+     * Attempts to create an {@link Attribute} according to the provided {@link AttributeDescriptor}
+     * whose value is represented by the given string {@param parsedValue}. Returns null if the string
+     * could not be parsed based on the descriptor's underlying {@link ddf.catalog.data.AttributeType.AttributeFormat}.
+     *
+     * @param attributeDescriptor The descriptor holding the format to use for parsing.
+     * @param parsedValue         The value to be parsed.
+     * @return A new {@link Attribute} if parsing succeeded, or null otherwise.
+     */
+    public Attribute createAttribute(AttributeDescriptor attributeDescriptor, String parsedValue) {
+        try {
+            if (attributeDescriptor.isMultiValued()) {
```

Comment:
Recommend doing some parameter checking here. To see if attributeDescriptor and parsedValue are null. Just looking to avoid potential NPEs in the future.